src/components/WPResponsiveImage/WPResponsiveImage.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Image } from '..';
+import { arrayFromRange, resizeWPImage } from '@quartz/js-utils';
+
+function WPResponsiveImage( {
+	alt,
+	aspectRatio,
+	fallbackWidth,
+	src,
+	widthRange,
+} ) {
+	const [ min, max ] = widthRange;
+	// Create an array of src widths based on the provided range
+	// Double the largest width to account for higher pixel density displays
+	const srcWidths = arrayFromRange( min, max * 2, 100 );
+
+	// Map source widths to image URLs
+	const srcSet = srcWidths
+		.map( width => `${resizeWPImage( src, width, width * aspectRatio )} ${width}w` )
```

Comment:
as a solve for "wordpress" issue, i would consider doing a regex on the URL in `resizeWPImage` and throwing an error if it doesn't match the expected hostname

Reply:
What do you think is a good thing to check for? `wp-content` maybe? I don't want to make the component unusable by people outside Quartz

Reply:
That's what the iOS app checks for: https://github.com/Quartz/ios/blob/8093cce9786728e2ccdef49953fd511da29c821f/Shared/Images/ImageSizeLimit.swift#L12

But thinking more about this now, I think a better solution would be to simply always optimistically add `w`, `h`, `width`, `height`, and `quality` params. Image hosting services may or may not support some subset of those but I think it's a bad idea to throw an error simply because an image URL doesn't match some regex test

Reply:
> I don't want to make the component unusable by people outside Quartz

I don't see how you can separate this component from the behavior of our media server, since its job is to automatically generate images at various sizes. I guess you could pass in the resize function? That seems messy and unintuitive.

We have or will have "manual" image components like `Image` and `ArtDirectedImage`. I think it's ok to have Quartz-specific components on top of them, as long as we label them as such in the docs. It's a UI library, but its ultimately our UI that we're implementing and no one else's.

Reply:
> I think it's a bad idea to throw an error simply because an image URL doesn't match some regex test

I definitely lean towards loud errors that break things instead of quiet errors that go unnoticed.